src/Experiment/src/ModuleBuilder.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Reflection.Emit.Experimental
+{
+    public class ModuleBuilder : System.Reflection.Module
+    {
+        private readonly Guid _s_guid;
+        public override System.Reflection.Assembly Assembly { get; }
+        public override string ScopeName { get; }
+
+        internal ModuleBuilder(string name, Assembly assembly)
+        {
+         //Example random GUID - I still need to understand when, how and why to generate them.
+        _s_guid= Guid.NewGuid();
+        ScopeName = name;
+            Assembly = assembly;
+        }
+
+        internal void AppendMetadata(MetadataBuilder metadataBuilder)
+        {
+            metadataBuilder.AddModule(
+                generation: 0,
+                metadataBuilder.GetOrAddString(ScopeName),
+                metadataBuilder.GetOrAddGuid(_s_guid),//See above comment on GUID
+                default(GuidHandle),
+                default(GuidHandle));
+        }
+
+        public void CreateGlobalFunctions()
+        {
+            throw new NotImplementedException();
+        }
+
+        // For all these methods, return type will be changed to "System.Reflection.Emit.Experiment.x" once non-empty modules/assemblies are supported.
+        public System.Reflection.Emit.EnumBuilder DefineEnum(string name, System.Reflection.TypeAttributes visibility, System.Type underlyingType) 
+        { 
+            throw new NotImplementedException(); 
+        }
```

Comment:
```suggestion
        public System.Reflection.Emit.EnumBuilder DefineEnum(string name, System.Reflection.TypeAttributes visibility, System.Type underlyingType) 
        => throw new NotImplementedException();
```
NIT: all of these can just be simplifyed to not have braces for now.